### ISSUE TITLE
error handling during load-lockfile

### ISSUE THREAD

**Issue by subsetpark** [janet-lang/jpm#16]
I'm encountering the following error when trying to call `load-lockfile` in a build:

```
sudo jpm --config-file=builds.sr.ht/config.janet load-lockfile
error: bad slot #1, expected string|symbol|keyword|buffer, got nil
  in string/find [src/core/string.c] on line 309
  in resolve-bundle-name [/usr/lib/janet/jpm/pm.janet] on line 88, column 11
  in resolve-bundle [/usr/lib/janet/jpm/pm.janet] on line 120, column 10
  in bundle-install [/usr/lib/janet/jpm/pm.janet] on line 178, column 4
  in load-lockfile [/usr/lib/janet/jpm/pm.janet] (tailcall) on line 245, column 5
  in _thunk [/usr/bin/jpm] on line -1, column -1
  in cli-main [boot.janet] on line 3618, column 39
```

I'm sure that it's because my `config.janet` is misconfigured, as the same command works fine on my machine.

However, the resulting error makes it difficult to tell what needs resolving. Perhaps we could program a little more defensively and surface an error message pointing to the problem?

**Comment by bakpakin** [comment]
Can you post the lockfile?

**Comment by subsetpark** [comment]
Sure: https://git.sr.ht/~subsetpark/bagatto/blob/master/lockfile.jdn

**Comment by bakpakin** [comment]
Should be addressed in latest master, 53e60388d5b1f16f505307286f5998d099b14fcc, verify and close @subsetpark

**Comment by subsetpark** [comment]
https://builds.sr.ht/~subsetpark/job/590269

`load-lockfile` works now! So my hypothesis was wrong, and this wasn't error handling, but an actual bug... which is now fixed :)
